### PR TITLE
fix: make sure that subscribe listener is triggered asynchronized

### PR DESCRIPTION
**Pull request by gxcsoccer**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/leader.js [hunk 1]
```diff
@@ -100,7 +100,9 @@ class Leader extends Base {
       send: {
         value(req) {
           const result = transcode.decode(req.data);
-          listener(result);
+          process.nextTick(() => {
+            listener(result);
+          });
```
Comment (gxcsoccer) [inline lib/leader.js]: 保证回调是异步触发的

Comment (fengmk2) [inline lib/leader.js]: 又是经典问题

**Comment by fengmk2** [comment]
1.5.3

**Comment by shaoshuai0102** [comment]
http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony

**Pull request merged**